File: Selenium/Project/Project9test.py

from selenium import webdriver 
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
import pytest 
from selenium.webdriver.support.ui import WebDriverWait 
from selenium.webdriver.support import expected_conditions as EC


@pytest.fixture 
def browser():
    # Initialize the Chrome WebDriver
    driver = webdriver.Chrome()
    yield driver
    # Quit the WebDriver
    driver.quit()

def test_search_job(browser):
        browser.get("https://alchemy.hguy.co/jobs/wp-admin")

        # Print the title of the page
        username_field = browser.find_element(By.ID, "user_login")  
        password_field = browser.find_element(By.ID, "user_pass") 
        login_button = browser.find_element(By.ID, "wp-submit")  

        # Enter login credentials
        username_field.send_keys("​root") 
        password_field.send_keys("pa$$w0rd")  

        # Submit the login form
        login_button.click()
        try:
            dashboard = WebDriverWait(browser, 10).until(EC.presence_of_element_located((By.CLASS_NAME, 'wp-menu-name')))
            assert dashboard.is_displayed(), "Login failed: Dashboard not displayed"
        except NoSuchElementException:
            assert False, "Login failed: Dashboard element not found"

        # Navigate to the job listing creation page
        browser.execute_script("window.scrollBy(0, 1000);")
        browser.implicitly_wait(5)
        create_job_button = browser.find_element(By.XPATH, "(//div[@class='wp-menu-name'])[6]")  # Adjust as needed
        create_job_button.click()

        #Click on Post a job
        post_job = browser.find_element(By.LINK_TEXT,"Add New")
        post_job.click()


        # Fill out the job listing form
        job_title = browser.find_element(By.ID, "post-title-0") 
        

        job_title.send_keys('Software Engineer')  # Replace with the job title
        

        # Submit the job listing
        submit_button = browser.find_element(By.CSS_SELECTOR, "a.components-button.editor-post-preview.is-secondary")  # Adjust the ID as needed
        submit_button.click()
        browser.implicitly_wait(5)

        #click on icon
        wordpress_button = browser.find_element(By.XPATH,"//div[@class='edit-post-header']/a")
        wordpress_button.click()

        #search for job
        jobsearch_button = browser.find_element(By.ID,"post-search-input")
        jobsearch_button.click()
        jobsearch_button.send_keys("Software Engineer")

        #click on search
        submitsearch_button = browser.find_element(By.ID,"search-submit")
        submitsearch_button.click()
        browser.implicitly_wait(5)

        #Verify
        verify_button = browser.find_element(By.XPATH,"(//div[@class='job_position'])[1]")
        verify_buttontext = verify_button.text
        assert 'Software Engineer' in verify_buttontext

        

if __name__ == '__main__':
    pytest.main()

File: Selenium/Project/Project9Test.java

package ProjectJob;

import java.time.Duration;
import java.util.Random;

import org.openqa.selenium.Alert;
import org.openqa.selenium.By;
import org.openqa.selenium.JavascriptExecutor;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Optional;
import org.testng.annotations.Parameters;
import org.testng.annotations.Test;

public class Project9Test {
    WebDriver driver;
    WebDriverWait wait;

    @BeforeClass
    public void beforeClass() {
        // Initialize the Firefox driver
        driver = new FirefoxDriver();
        // Initialize the wait object
        wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        // Open browser
        driver.get("https://alchemy.hguy.co/jobs/wp-admin");
    }

    @Test
    @Parameters({ "username1", "password1", "message" })
    public void loginTestCase(String username1, String password1, @Optional("Login Success!") String message) {
        // Find username and password fields
        WebElement usernameField = driver.findElement(By.id("user_login"));
        WebElement passwordField = driver.findElement(By.id("user_pass"));

        // Enter credentials
        usernameField.sendKeys(username1);
        passwordField.sendKeys(password1);

        // Click the submit button
        driver.findElement(By.xpath("//input[@class='button button-primary button-large']")).click();

       

        // Assert login message
        String loginMessage = driver.findElement(By.xpath("//h1")).getText();
        Assert.assertEquals("Dashboard", loginMessage);
        
    }
    
    
    @Test(priority = 1)
    public void postAJob() {
    	JavascriptExecutor js = (JavascriptExecutor) driver;
    	js.executeScript("window.scrollBy(0,1000)");
    	
    	WebElement hamburgerMenu = driver.findElement(By.xpath("(//div[@class='wp-menu-name'])[6]")); 
    	
    	hamburgerMenu.click(); 
    	wait = new WebDriverWait(driver, Duration.ofSeconds(10));
   
    	WebElement submenuItem = driver.findElement(By.linkText("Add New"));
    	submenuItem.click();
    
    	wait = new WebDriverWait(driver, Duration.ofSeconds(120));
    	WebElement jobTitle1 = driver.findElement(By.cssSelector("textarea.editor-post-title__input"));
    	String jobCharacters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
    	StringBuilder jobTitle = new StringBuilder(); 
    	Random random1 = new Random(); 
    	for (int i = 0; i < 10; i++) 
    	{ // Generate random string of 10 characters 
    		jobTitle.append(jobCharacters.charAt(random1.nextInt(jobCharacters.length()))); 
    		} 
    	String AppendText = "TestQA_";
    	String finalJobTitle = AppendText + jobTitle;
    	String jobTitle3 = finalJobTitle.toString();
    	
    	jobTitle1.sendKeys(jobTitle3);
		/*
		 * //Email id String characters =
		 * "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
		 * StringBuilder email = new StringBuilder(); Random random = new Random(); for
		 * (int i = 0; i < 10; i++) { // Generate random string of 10 characters
		 * email.append(characters.charAt(random.nextInt(characters.length()))); }
		 * email.append("@example.com"); String emailInput = email.toString();
		 * 
		 * WebElement emailLocator = driver.findElement(By.id("create_account_email"));
		 * emailLocator.sendKeys(emailInput);
		 */
    	//company Website
    	WebElement companyWebsite = driver.findElement(By.name("_company_website"));
    	companyWebsite.click();
    	companyWebsite.sendKeys("ibm.com");
    	
    	//Company Name
    	WebElement companyName = driver.findElement(By.name("_company_name"));
    	companyName.click();
    	companyName.sendKeys("IBM");
    	
    	
    	//Publish
    	WebElement Publish = driver.findElement(By.cssSelector("a.components-button.editor-post-preview.is-secondary"));
    	Publish.click();
    	wait = new WebDriverWait(driver, Duration.ofSeconds(120));
    	
    	
		
		/*
		 * WebElement viewJob = driver.findElement(By.linkText("View Job"));
		 * viewJob.click(); wait = new WebDriverWait(driver, Duration.ofSeconds(30));
		 * WebElement verifyJobTitle = driver.findElement(By.xpath("//h1"));
		 */
		
		
		  //Navigating back to home page 
    	WebElement homePage = driver.findElement(By.xpath("//a[@aria-label='View Posts']"));
		homePage.click(); 
		wait = new WebDriverWait(driver, Duration.ofSeconds(30));
		WebElement verifyJobTitle = driver.findElement(By.xpath("(//a[@class='tips job_title'])[1]"));
		 
    	String actualJobTitle = verifyJobTitle.getText();
    	System.out.println("Jobs Page Title is: " + actualJobTitle);
        
    	
    	
    
        // Assert page title
        Assert.assertEquals(actualJobTitle, jobTitle3);
    }

    @AfterClass
    public void afterClass() {
        // Close browser
        driver.quit();
    }
}
